src/svm/svc.rs

```diff
@@ -84,12 +85,107 @@ use serde::{Deserialize, Serialize};
 
 use crate::api::{PredictorBorrow, SupervisedEstimatorBorrow};
 use crate::error::{Failed, FailedError};
-use crate::linalg::basic::arrays::{Array1, Array2, MutArray};
+use crate::linalg::basic::arrays::ArrayView1;
+use crate::linalg::basic::arrays::{Array, Array1, Array2, MutArray};
+use crate::linalg::basic::matrix::DenseMatrix;
 use crate::numbers::basenum::Number;
 use crate::numbers::realnum::RealNumber;
 use crate::rand_custom::get_rng_impl;
 use crate::svm::Kernel;
 
+pub struct MultiClassSVCParameters<'a> {
+    xs_filtered: Vec<DenseMatrix<f64>>,
```

Comment:
this needs to be implemented using the generic `TX` `TY`. It should work for all supported types, not just f64.